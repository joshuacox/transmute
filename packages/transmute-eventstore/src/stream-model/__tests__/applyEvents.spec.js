const TransmuteEventStore = require('../../index');
const transmuteConfig = require('../../../../../transmute-config');
const eventStoreArtifact = require('../../../build/contracts/EventStore.json');

const eventStore = new TransmuteEventStore({
  eventStoreArtifact,
  ...transmuteConfig
});

const StreamModel = require('../index');

const events = require('../../__mock__/events.json');

describe('applyEvents', () => {
  beforeAll(async () => {
    await eventStore.init();
  });
  it('idempotent', async () => {
    const filter = event => {
      return true;
    };
    const reducer = (state, event) => {
      return state;
    };
    const streamModel = new StreamModel(eventStore, filter, reducer);
    streamModel.applyEvents(events);
    expect(streamModel.state.contractAddress).toEqual(
      eventStore.eventStoreContractInstance.address
    );
  });

  it('respects filter', async () => {
    const filter = event => {
      return event.key.type === 'patient' && event.key.id === '0';
    };
    const reducer = (state, event) => {
      return {
        ...state,
        events: [...(state.events || []), event]
      };
    };
    const streamModel = new StreamModel(eventStore, filter, reducer);
    streamModel.applyEvents(events);
    expect(streamModel.state).toMatchSnapshot();
  });
});
