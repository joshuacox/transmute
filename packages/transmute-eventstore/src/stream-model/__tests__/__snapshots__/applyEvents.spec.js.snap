// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyEvents respects filter 1`] = `
Object {
  "contractAddress": "0x5c3ebf0fdb3176dfe8ffde603bcfccf7e33fad7e",
  "lastIndex": "4",
  "model": Object {
    "events": Array [
      Object {
        "index": "1",
        "key": Object {
          "id": "0",
          "type": "patient",
        },
        "value": Object {
          "type": "USER_REGISTERED",
          "username": "bob@example.com",
        },
      },
      Object {
        "index": "3",
        "key": Object {
          "id": "0",
          "type": "patient",
        },
        "value": Object {
          "guid": "0x0...",
          "type": "DEVICE_LINKED",
        },
      },
      Object {
        "index": "4",
        "key": Object {
          "id": "0",
          "type": "patient",
        },
        "value": Object {
          "type": "USER_UPDATED",
          "username": "bob2@example.com",
        },
      },
    ],
  },
}
`;
